packages/client/src/objectSet/ObjectSet.test.ts

```diff
@@ -436,6 +437,260 @@ describe("ObjectSet", () => {
     });
   });
 
+  describe("Derived Properties Object Set", () => {
+    it("does not allow aggregate or selectProperty before a link type is selected", () => {
+      client(Employee).withProperties({
+        "derivedPropertyName": (base) =>
```

Comment:
Can we also add a test to make sure that within a WithPropertiesClause you can't just return something like `base.pivotTo("lead")` without doing some sort of selection or aggregation?

Reply:
Apologies in advance if I missed this somewhere, but don't seem to see that being tested anywhere yet

Reply:
This should be guarded by the type of the lambda, but I will add an explicit test for this behavior